lca-cli/ibi-preparation/ibipreparation.go

```diff
@@ -81,15 +92,28 @@ func (i *IBIPrepare) Run() error {
 	return i.shutdownNode()
 }
 
-func (i *IBIPrepare) precacheFlow(imageListFile string) error {
-	// TODO: add support for mirror registry
+func (i *IBIPrepare) precacheFlow(imageListFile, seedInfoFile string) error {
 	if i.config.PrecacheDisabled {
 		i.log.Info("Precache disabled, skipping it")
 		return nil
 	}
 
-	i.log.Info("Precaching imaging")
-	imageList, err := prep.ReadPrecachingList(imageListFile, "", "", false)
+	i.log.Info("Checking seed image info")
+	seedInfo, err := seedclusterinfo.ReadSeedClusterInfoFromFile(seedInfoFile)
+	if err != nil {
+		return fmt.Errorf("failed to read seed info: %s, %w", common.PathOutsideChroot(seedInfoFile), err)
+	}
+	i.log.Info("Collected seed info for precache: ", fmt.Sprintf("%+v", seedInfo))
+
+	i.log.Info("Checking whether to override seed registry")
+	shouldOverrideSeedRegistry, err := i.shouldOverrideSeedRegistry(seedInfo.MirrorRegistryConfigured, seedInfo.ReleaseRegistry)
+	if err != nil {
+		return fmt.Errorf("failed to check ShouldOverrideSeedRegistry %w", err)
+	}
+	i.log.Info("Should override seed registry", "shouldOverrideSeedRegistry", shouldOverrideSeedRegistry, "releaseRegistry", i.config.ReleaseRegistry)
```

Comment:
Can we share the code with https://github.com/openshift-kni/lifecycle-agent/blob/main/controllers/prep_handlers.go#L377-L398 ?

Reply:
That's what I first thought, but the main difference is where we fetch the mirror registry locations from.

- IBU - we're fetching the mirror registry locations from the [cluster ](https://github.com/openshift-kni/lifecycle-agent/blob/main/utils/client_helper.go#L291-L314)
- IBI (what's implemented here) we're fetching the mirror registry locations from `/etc/containers/registries.conf` - https://github.com/openshift-kni/lifecycle-agent/pull/727/files#diff-2a78eaecaa85ed29707e26c8abe068e48b486f8c3f2b816e7f9135a22f890441R260

Perhaps I could try refactoring the code to share some logic. WDYT?

Reply:
If it's a lot of trouble in order to share 3 lines of code you can resolve this comment

Reply:
I think it's just 4 lines of codes we're sharing, but it makes sense to share them anyway https://github.com/openshift-kni/lifecycle-agent/pull/727/files#diff-89d0ea845042a03c3156e8b2e5d8e1980f1b97379fd4a26d02d5c271cc18195fR414-R422.

Thanks for the suggestion :pray: